docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md

```diff
@@ -0,0 +1,80 @@
+---
+id: query-editor
+title: The query editor
+description: Testing the query editor
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - testing
+  - e2e
+  - data-source
+  - query editor
+sidebar_position: 20
+---
+
+## Introduction
+
+The query editor is a central piece of a data source plugin as this is where the query that is used to fetch data is being crafted. Some data source plugins provide rich query editors with various query types targeting different APIs. They can have visual query builders and/or code editors with support for advanced features such as intellisense and autocompletion.
+
+### Testing parts of the query editor in isolation
+
+The query editor in the following example loads regions from a remote API and filters out any GovCloud regions before presenting them in a dropdown. To test that the filtering is working as expected, we can mock the regions resource response and assert that only the non GovCloud regions are shown when the regions dropdown is clicked.
```

Comment:
```suggestion
The query editor in the following example loads regions via a request to `/regions` and filters out regions containing `-gov-` before presenting them in a dropdown. 

To test that the filtering is working as expected, we can mock the `/regions` resource response and assert that only the regions without `-gov-` in their name are shown when the regions dropdown is clicked.
```

Reply:
>We can omit specifics such as GovCloud (I don't know what they are but I can tell by the name is about having -gov-)

Yeah good point - it's irrelevant for this article, so let's remove that part about govcloud. 

>My main concern is this example assumes a lot of the knowledge of the person about mocking datasources request. Maybe we can link (if we have) the article to mock datasource requests?

We don't have such an article yet, but I added a sentence that explains what this method does. My changes are based on your suggestions. Let me know what you think.